content/en/docs/concepts/storage/storage-classes.md

```diff
@@ -415,6 +415,33 @@ This internal provisioner of OpenStack is deprecated. Please use [the external c
 
 ### vSphere
 
+There are two types of provisioners for vSphere storage classes: 
+
+- [CSI provisioner](#csi-provisioner): `csi.vsphere.vmware.com`
+- [vCP provisioner](#vcp-provisioner): `kubernetes.io/vsphere-volume`
+
+In-tree provisioners are [deprecated](https://kubernetes.io/blog/2019/12/09/kubernetes-1-17-feature-csi-migration-beta/#why-are-we-migrating-in-tree-plugins-to-csi). For more information on the CSI provisioner, see [Kubernetes vSphere CSI Driver](https://vsphere-csi-driver.sigs.k8s.io/) and [vSphereVolume CSI migration](https://kubernetes.io/docs/concepts/storage/volumes/#csi-migration-5).
+
+#### CSI Provisioner {#vsphere-provisioner-csi}
```

Comment:
In general, we have not been documenting CSI storageclasses in the kubernetes docs since these types are not part of core Kubernetes.

Reply:
OK, thanks @msau42 . To clarify, the proposed sections should not be included?

Reply:
I think the rest of the sections are fine, but maybe for the CSI example, we should just point to the CSI repo for examples.

Reply:
@msau42 – I agree, the entire part of taking these out of tree was to do things like not document them as a part of core :) 

I notice that https://kubernetes-csi.github.io/docs/drivers.html exists – is the plan to move some/all of the docs on this github.io subsite into k/website at some point in the future?

Reply:
The csi docs intended audience is for driver developers, not really end users, whereas Kubernetes docs are intended for concepts and end users. But for csi-driver specific functionality, we have been pointing directly to a csi driver's repo for end user documentation.

Reply:
Given how CSI works, is it feasible to host the documentation about vendor-specific CSI drivers on the vendor's site, and leat web search engines make that discoverable?

We can't rely on all CSI drivers being in-tree or in-project, quite the opposite in fact.

Reply:
Yes, the drivers page at https://kubernetes-csi.github.io/docs/drivers.html is intended to be a listing with links to the csi driver's specific instructions

Reply:
I would be more than happy to link, but the sites https://github.com/kubernetes-sigs/vsphere-csi-driver and https://kubernetes-csi.github.io/docs/drivers.html do not have storage class creation examples. They provide examples for implementing the driver(s). We need to provide a straightforward example of how to create a storage class using the specified CSI, just like was done for in-tree provisioners. Creating the storage class is within the purview of the Kubernetes user workflow. This is how I ran into the issue.

Reply:
I would be more than happy to link, but the sites https://github.com/kubernetes-sigs/vsphere-csi-driver and https://kubernetes-csi.github.io/docs/drivers.html do not have storage class creation examples. They provide examples for implementing the driver(s). We need to provide a straightforward example of how to create a storage class using the specified CSI, just like was done for in-tree provisioners. Creating the storage class is within the purview of the Kubernetes user workflow. This is how I ran into the issue.

Reply:
https://github.com/kubernetes-sigs/vsphere-csi-driver is the right place to put driver-specific documentation and examples of how to use it. cc @xing-yang to help coordinate

Reply:
I removed the example and added a direct link to an example in the repo (which I was able to locate). Thanks.